CMakeLists.txt

```diff
@@ -1,68 +1,52 @@
 cmake_minimum_required(VERSION 3.7)
 project(engine C CXX)
+
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS False)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/modules")
 
-set(PHYSFS_BUILD_STATIC ON)
-set(PHYSFS_BUILD_SHARED OFF)
-set(PHYSFS_BUILD_TEST OFF)
-add_subdirectory("extern/physfs")
-
-set(ASSIMP_BUILD_NO_EXPORT ON)
-set(ASSIMP_BUILD_ALL_IMPORTERS_BY_DEFAULT FALSE)
-
-set(ASSIMP_BUILD_OBJ_IMPORTER ON)
-set(ASSIMP_BUILD_FBX_IMPORTER ON)
-set(ASSIMP_BUILD_GLTF_IMPORTER ON)
-set(ASSIMP_BUILD_PLY_IMPORTER ON)
-set(ASSIMP_BUILD_BLEND_IMPORTER ON)
-set(ASSIMP_BUILD_COLLADA_IMPORTER ON)
-
-set(ASSIMP_BUILD_ASSIMP_TOOLS OFF)
-set(ASSIMP_BUILD_TESTS OFF)
-add_subdirectory("extern/assimp")
-
-find_package(SDL2 REQUIRED)
-
-include_directories(
-	${SDL2_INCLUDE_DIRS}
-	${ASSIMP_INCLUDE_DIR}
-	${CMAKE_SOURCE_DIR}/extern/physfs/src
-	${CMAKE_SOURCE_DIR}/extern/assimp/include
+include(extern/extern.cmake)
+
+file(GLOB HEADERS
```

Comment:
Splitting globs up into headers/source/shaders may not matter right this second, but it often a helpful thing to have.